cirq-core/cirq/experiments/z_phase_calibration.py

```diff
@@ -283,3 +284,78 @@ def plot_z_phase_calibration_result(
         ax.set_title('-'.join(str(q) for q in pair))
         ax.legend()
     return axes
+
+
+@transformer_api.transformer
+class CalibrationTransformer:
+
+    def __init__(
+        self,
+        target: 'cirq.Gate',
+        calibration_map: Dict[Tuple['cirq.Qid', 'cirq.Qid'], 'cirq.PhasedFSimGate'],
+    ):
+        """Create a CalibrationTransformer that replaces gates matching `target`.
+
+        Args:
+            target: The target gate. Any gate matching this
+                will be replaced based on the content of `calibration_map`.
+            calibration_map:
+                A map mapping qubit pairs to calibrated gates. This is the output of
+                calling `calibrate_z_phases`.
+        """
+        self.target = target
+        if isinstance(target, ops.PhasedFSimGate):
+            self.target_as_fsim = target
+        elif (gate := ops.PhasedFSimGate.from_matrix(protocols.unitary(target))) is not None:
+            self.target_as_fsim = gate
+        else:
+            raise ValueError(f"{target} is not equivalent to a PhasedFSimGate")
+        self.calibration_map = calibration_map
+
+    def __call__(
+        self,
+        circuit: 'cirq.AbstractCircuit',
+        *,
+        context: Optional[transformer_api.TransformerContext] = None,
+    ) -> 'cirq.Circuit':
+        """Replace every occurance of a calibrated gate with a proper replacement.
+
+        Args:
+            circuit: Circuit to transform.
+            context: Optional transformer context (not used).
+
+        Returns:
+            New circuit with each gate matching `target` and acting on a pair
+                in `calibration_map` replaced by the gate that cancels the effect of z-phases.
+        """
+        new_moments = []
+        for moment in circuit:
+            new_moment = []
+            for op in moment:
+                if op.gate != self.target:
+                    # not a target.
+                    new_moment.append(op)
+                    continue
+                assert len(op.qubits) == 2
+                gate = self.calibration_map.get(op.qubits, None) or self.calibration_map.get(
+                    op.qubits[::-1], None
+                )
+                if gate is None:
+                    # no calibrated version exist
+                    new_moment.append(op)
+                    continue
+                delta_theta = gate.theta - self.target_as_fsim.theta
+                delta_phi = gate.phi - self.target_as_fsim.phi
+                delta_chi = gate.chi - self.target_as_fsim.chi
+                delta_zeta = gate.zeta - self.target_as_fsim.zeta
+                delta_gamma = gate.gamma - self.target_as_fsim.gamma
+                new_gate = ops.PhasedFSimGate(
+                    theta=self.target_as_fsim.theta - delta_theta,
+                    gamma=self.target_as_fsim.gamma - delta_gamma,
+                    phi=self.target_as_fsim.phi - delta_phi,
```

Comment:
`theta` and `phi` are not z-phases. Can we directly insert the z-rotations that take care of cancelling the phases (just the other three angles)?

Reply:
> `theta` and `phi` are not z-phases. 
yes, but if they were calibrated then it makes sense to change them as well. if they were not calibrated then the delta would be $=0$ and nothing happens.

> Can we directly insert the z-rotations that take care of cancelling the phases (just the other three angles)?
we can, I'm okay either way

Reply:
In order for this to be useful on hardware, it would be great if there were at least an option to only cancel the z-phases (not theta and phi) and for this to be implemented as z rotations rather than fSim gates. Thanks, Nour.

Reply:
ptal